### ISSUE TITLE
add purge_routes option to ec2_vpc_route_table

### ISSUE THREAD

**Issue by therc** [ansible/ansible-modules-extras#1746]
##### ISSUE TYPE
- Bug Report
- Feature Idea
##### Plugin Name:

ec2_vpc_route_table
##### Ansible Version:

```
ansible 2.0.0.2
```
##### Ansible Configuration:

No tweaks
##### Environment:

Ubuntu 15.10
##### Summary:

From #1511:

Could we allow for routes not to get deleted, if asked?

ec2_group has options purge_rules and purge_rules_egress, which default to true. If you set them to false, the module won't delete existing rules that are not found in the fields.

Use case for this: you create a new route table for Kubernetes. Kubernetes will then add a new route for each node's own subnet as they come online. You try to rerun your playbook and the nodes become unreachable because Ansible has deleted the routes (Kubernetes will still recreate them, but it might take ten seconds). Having a `purge_routes` that I could set to false would help.

**Comment by akvadrako** [comment]
This would be nice to have since VPC peerings must be maintained manually, it makes sense to also maintain routes between peered VPCs manually.

**Comment by ansibot** [comment]
@erydo, @wimnat, ping. This issue is still waiting on your response.
[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by ansibot** [comment]
@erydo, @wimnat, ping. This issue is still waiting on your response.
[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by stepanstipl** [comment]
I just had similar problem, so PR #2959 - added `purge_routes` option.

@therc @akvadrako Can you guys check that this does what you were asking for?

**Comment by ansibot** [comment]
@erydo, @wimnat, ping. This issue is still waiting on your response.
[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by ansibot** [comment]
@erydo, @wimnat, ping. This issue is still waiting on your response.
[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by ghost** [comment]
@stepanstipl. Just came across this problem myself in something that I'm writing. Looks like PR #2959 works from my testing so far. Though I can't say that's full testing yet.

**Comment by stepanstipl** [comment]
@jm-oss Thanks, good to hear. It would be good to get this merged.

**Comment by ansibot** [comment]
This repository has been locked. All new issues and pull requests should be filed in https://github.com/ansible/ansible

Please read through the <a href="http://docs.ansible.com/ansible/dev_guide/repomerge.html#move-issues-and-prs-to-new-repo">repomerge</a> page in the dev guide. The guide contains links to tools which automatically move your issue or pull request to the ansible/ansible repo.
